python/dpu_utils/utils/iterators.py

```diff
@@ -188,3 +189,27 @@ def __next__(self):
             self.__worker_process_outer.join()
             raise StopIteration
         return next_element
+
+
+def shuffled_iterator(input: Iterator[T], buffer_size: int = 10000, out_slice_sizes: int = 500) -> Iterator[T]:

```

Comment:
`input` -> `input_iterator` might be good, to avoid conflict with Python's `input()`